src/cider/nrepl.clj

```diff
@@ -25,8 +29,9 @@
         sym (symbol (name `~fn-name))]
     (swap! DELAYS assoc sym
            (delay
-             (require `~ns)
-             (resolve-or-fail `~fn-name)))
+             (locking LOCK
```

Comment:
I wonder if we can't serialize this in some more elegant manner. This combination of `delay` + `locking` seems very unclojury to me.

Reply:
Not sure what is troubling you. Those are rarely used clojure functions, but they do precisely what's needed in that context. Anything else would likely imply more code. Do you have some concrete ideas?